app/src/controllers/plugins/reducer.js

```diff
@@ -118,9 +119,19 @@ export const integrationsReducer = (state = {}, { type, payload }) => {
   }
 };
 
+export const loadingPluginsReducer = (state = false, { type, payload }) => {
+  switch (type) {
+    case SET_PLUGINS_LOADING:
+      return payload;
+    default:
+      return state;
+  }
+};
+
 export const pluginsReducer = combineReducers({
   plugins: queueReducers(fetchReducer(NAMESPACE), updatePluginLocallyReducer),
   publicPlugins: fetchReducer(PUBLIC_PLUGINS),
   integrations: integrationsReducer,
   uiExtensions: uiExtensionsReducer,
+  pluginsLoading: loadingPluginsReducer,
```

Comment:
Please use `loadingReducer(NAMESPACE)` instead. Loading state will be updated accordingly withob `fetchDataAction`.